src/schema/users.ts

```diff
@@ -1042,6 +1083,54 @@ export const resolvers: IResolvers<any, Context> = {
         });
       }
 
+      return { _: true };
+    },
+    acceptFeatureInvite: async (
+      _,
+      {
+        token,
+        referrerId,
+        feature,
+      }: {
+        token: string;
+        referrerId: string;
+        feature: string;
+      },
+      ctx: Context,
+    ): Promise<GQLEmptyResponse> => {
+      const referrerInvite = await ctx.con
+        .getRepository(Invite)
+        .findOneByOrFail({
+          userId: referrerId,
+          token: token,
+          campaign: feature as CampaignType,
+        });
+
+      if (referrerInvite.count >= referrerInvite.limit) {
+        throw new ValidationError(SubmissionFailErrorKeys.InviteLimitReached);
+      }
+
+      await ctx.con.transaction(async (entityManager): Promise<void> => {
+        try {
+          await entityManager.getRepository(Feature).insert({
```

Comment:
Do we want an additional check here to make sure that only if if the feature is search we generate the invite? Or do we leave it generic?

Reply:
Generic is cool!